tests/core/toolkit/test_destroyable_thread.py

```diff
@@ -38,11 +38,22 @@ def infinite_wait():
             time.sleep(0.1)
```

Comment:
While we're here it might be nice to make this a `time.sleep(0.001)` to make the test a bit more snappy?